pyadjoint/tape.py

```diff
@@ -160,11 +160,13 @@ class Tape(object):
 
     """
     __slots__ = ["_blocks", "_tf_tensors", "_tf_added_blocks", "_nodes",
-                 "_tf_registered_blocks", "_bar", "_package_data"]
+                 "_tf_registered_blocks", "_bar", "_package_data",
+                 "_checkpoint_manager", "latest_checkpoint",
+                 "_eagerly_checkpoint_outputs"]
 
     def __init__(self, blocks=None, package_data=None):
         # Initialize the list of blocks on the tape.
-        self._blocks = [] if blocks is None else blocks
+        self._blocks = TimeStepSequence(blocks=blocks if blocks else ())
```

Comment:
```suggestion
        self._blocks = TimeStepSequence(blocks=blocks if blocks else [])
```
Feels wrong to send in an empty tuple, as they are immutable, and gives the idea that one cannot append blocks at a later stage.